packages/@react-spectrum/s2/style/style-macro.ts

```diff
@@ -115,14 +115,40 @@ interface MacroContext {
   addAsset(asset: {type: string, content: string}): void
 }
 
+// djb2 hash function.
+// http://www.cse.yorku.ca/~oz/hash.html
+function hash(v: string) {
+  let hash = 5381;
+  for (let i = 0; i < v.length; i++) {
+    hash = ((hash << 5) + hash) + v.charCodeAt(i) >>> 0;
+  }
+  return hash;
+}
+
+function generateThemeHash(theme: Theme): string {
+  let themeString = JSON.stringify(theme, (_, value) => {
+    if (typeof value === 'function') {
+      return value.toString();
```

Comment:
I guess a problem here would be that many of the functions will be the same, just with different arguments. For example, `createColorProperty` always returns a function that will stringify to the same thing, but it uses different values from its closure scope...